src/node_internals.h

```diff
@@ -75,6 +75,100 @@ struct sockaddr;
+ * in node::Start(), need to add built-in modules in the following list.
+ * Then in node::RegisterBuiltinModules(), it calls modules' registration
+ * function. This helps the built-in modules are loaded properly when
+ * node is built as static library. No need to depends on the
+ * __attribute__((constructor)) like mechanism in GCC.
+ */
+#define NODE_BUILTIN_MODULES(V)                                               \
+    V(async_wrap,     MODREG_WITH_DEF)                                        \
+    V(cares_wrap,     MODREG_WITH_DEF)                                        \
+    V(fs_event_wrap,  MODREG_WITH_DEF)                                        \
+    V(js_stream,      MODREG_WITH_DEF)                                        \
+    V(module_wrap,    MODREG_WITH_DEF)                                        \
+    V(buffer,         MODREG_WITH_DEF)                                        \
+    V(config,         MODREG_WITH_DEF)                                        \
+    V(contextify,     MODREG_WITH_DEF)                                        \
+    V(fs,             MODREG_WITH_DEF)                                        \
+    V(http2,          MODREG_WITH_DEF)                                        \
+    V(http_parser,    MODREG_WITH_DEF)                                        \
+    V(os,             MODREG_WITH_DEF)                                        \
+    V(performance,    MODREG_WITH_DEF)                                        \
+    V(serdes,         MODREG_WITH_DEF)                                        \
+    V(url,            MODREG_WITH_DEF)                                        \
+    V(util,           MODREG_WITH_DEF)                                        \
+    V(v8,             MODREG_WITH_DEF)                                        \
+    V(zlib,           MODREG_WITH_DEF)                                        \
+    V(icu,            ICU_MODREG_DEF)                                         \
+    V(pipe_wrap,      MODREG_WITH_DEF)                                        \
+    V(process_wrap,   MODREG_WITH_DEF)                                        \
+    V(signal_wrap,    MODREG_WITH_DEF)                                        \
+    V(spawn_sync,     MODREG_WITH_DEF)                                        \
+    V(stream_wrap,    MODREG_WITH_DEF)                                        \
+    V(tcp_wrap,       MODREG_WITH_DEF)                                        \
+    V(timer_wrap,     MODREG_WITH_DEF)                                        \
+    V(tty_wrap,       MODREG_WITH_DEF)                                        \
+    V(udp_wrap,       MODREG_WITH_DEF)                                        \
+    V(uv,             MODREG_WITH_DEF)                                        \
+    V(inspector,      MODREG_WITH_DEF)                                        \
+    V(crypto,         OPENSSL_MODREG_DEF)                                     \
+    V(tls_wrap,       OPENSSL_MODREG_DEF)
```

Comment:
This is not wrong but I'd approach it a little differently.
```c++
#define NODE_BUILTIN_STANDARD_MODULES(V) \
  V(async_wrap) \
  V(cares_wrap) \
  // ...

#if HAVE_OPENSSL
#define NODE_BUILTIN_OPENSSL_MODULES(V) V(crypto) V(tls_wrap)
#else
#define NODE_BUILTIN_OPENSSL_MODULES(V)
#endif

#if NODE_HAVE_I18N_SUPPORT
#define NODE_BUILTIN_ICU_MODULES(V) V(icu)
#else
#define NODE_BUILTIN_ICU_MODULES(V)
#endif

#define NODE_BUILTIN_MODULES(V) \
  NODE_BUILTIN_STANDARD_MODULES(V) \
  NODE_BUILTIN_OPENSSL_MODULES(V) \
  NODE_BUILTIN_ICU_MODULES(V)
```
It's arguably a little easier to comprehend because the V macro takes only one argument this way.